tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestConstSimplifier.scala

```diff
@@ -277,18 +283,25 @@ class TestConstSimplifier extends AnyFunSuite with BeforeAndAfterEach with Check
     check(prop, minSuccessful(1000), sizeRange(8))
   }
 
+  // TODO: remove and revisit tests below once all builder methods are implemented
+  import TypedPredefs._
+  import scala.language.implicitConversions
+  implicit def builderBridge(bi: TBuilderInstruction): BuilderTlaExWrapper =
+    BuilderTlaExWrapper(build(bi))
+
   test("simplifies logical expressions that result in part of the expression") {
-    val prop = forAll(gens.genTlaEx(ops)(gens.emptyContext)) { ex =>
-      val expressions = List(
-          tla.and(ex).as(BoolT1()),
-          tla.and(ex, tla.bool(true).as(BoolT1())).as(BoolT1()),
-          tla.or(ex).as(BoolT1()),
-          tla.or(ex, tla.bool(false).as(BoolT1())).as(BoolT1()),
-          tla.ite(tla.bool(true).as(BoolT1()), ex, tla.bool(false).as(BoolT1())).as(BoolT1()),
-          tla.ite(tla.bool(false).as(BoolT1()), tla.bool(false).as(BoolT1()), ex).as(BoolT1()),
-          tla.ite(ex, tla.bool(true).as(BoolT1()), tla.bool(false).as(BoolT1())).as(BoolT1()),
-          tla.ite(ex, ex, ex).as(BoolT1()),
-          tla.not(tla.not(ex).as(BoolT1())).as(BoolT1()),
+    val prop = forAll(gens.genTlaEx(ops)(gens.emptyContext)) { plainEx =>
+      val ex = tla.useTrustedEx(plainEx.withTag(Typed(BoolT1())))
+      val expressions: List[TlaEx] = List(
+          tla.and(ex),
+          tla.and(ex, tla.bool(true)),
+          tla.or(ex),
+          tla.or(ex, tla.bool(false)),
+          oldTla.ite(tla.bool(true), plainEx, tla.bool(false)).as(BoolT1()),
+          oldTla.ite(tla.bool(false), tla.bool(false), plainEx).as(BoolT1()),
+          oldTla.ite(plainEx, tla.bool(true), tla.bool(false)).as(BoolT1()),
+          oldTla.ite(plainEx, ex, ex).as(BoolT1()),
```

Comment:
why do you need `oldTla`? Is it due to `ite` not being implemented?

Reply:
Yes, that's why the takeaway from this POC is that we should implement all methods first.